Dockerfile

```diff
@@ -1,27 +1,84 @@
-FROM node:10
+# This Docker file tends to be used only
+# for `development` and `test` stage
+#
+# CLI command to run this file: script="some-name" docker-compose up --build
+# `some-name` is one of the names provided under `scripts` tag in `package.json` file
+# example: $ script="test" docker-compose up --build
+# default: $ docker-compose up --build      --> will run telescope with debug-server script
+
+
+# Dockerfile
+
+# To guarantee the same stack for each image build:
+# - Specify Ubuntu as the parent image
+# - Install Node.js to let using npm in this stage of the build
+# - Use Redis image via compose-file
+# comment: The Ubuntu and Node.js versions need to be selected carefully (important: https://github.com/nodesource/distributions),
+#   and can not be passed as .env variables or arguments.
+
+# UBUNTU VERSION:
+#   Using a stable version with standard support up to April 2023
+#   https://wiki.ubuntu.com/Releases, https://hub.docker.com/_/ubuntu/
+ARG UBUNTU_VERSION=18.04
+
+# NODE VERSION:
+#   Using an stable version rather than the latest
+#   Node version 12, https://nodejs.org/en/about/releases/, https://hub.docker.com/_/node/
+#   End of life for this version: 2022-04-30 subject to change
+ARG NODE_VERSION=12
 
-# Change working directory
-WORKDIR "/telescope"
 
-# Update packages and install dependency packages for services
-RUN apt-get update \
- && apt-get dist-upgrade -y \
- && apt-get clean \
- && echo 'Finished installing dependencies'
+# Specify parent image to build this image
+FROM ubuntu:$UBUNTU_VERSION
 
-# Copy package.json
+# Update OS
+# Install Linux `apt-utils` package to enhance packages configuration
+# Install Linux `bc` package to support interactive execution of statements
+# Install Linux `inotify-tools` to support nodemon file change tracking
+# Install Linux `curl` package to support transfer data from URL
+RUN apt-get update -y
+RUN apt-get install -y apt-utils
+RUN apt-get install -y bc
+RUN apt-get install -y inotify-tools
+RUN apt-get install -y curl
+RUN echo "Finished installing OS dependencies"
```

Comment:
Let's get rid of `nodemon`, we don't need it in production, and if it's requiring us to pull in all sorts of OS deps, it's not what we want here.

Reply:
😢 it took me days to figure out how to have container aware of the changes for `nodemon` , and now it's like my child 😃. 
Sure, I will remove everything related to it.

Reply:
I'm breaking everyone's heart this week, sorry :(  But on the bright side, think of all the amazing learning you've done! 

I think dumping this is the better solution, since it's not really needed, it just got added when the web server went in.

Reply:
That's true. It was obviously a lot of learning for me rather than me being helpful for the project.